### PR TITLE
Tl/vite

### PR DESCRIPTION
**Pull request by tjlav5**

### CODE REVIEW

File: src/components/Firestore/FirestoreEmulatedApiProvider.tsx [hunk 1]
```diff
@@ -35,17 +48,18 @@ const FIRESTORE_OPTIONS = {};
  * the Emulator Hub.
  */
 export const FirestoreEmulatedApiProvider: React.FC<
-  React.PropsWithChildren<{
-    disableDevTools?: boolean;
-  }>
-> = React.memo(({ children, disableDevTools }) => {
+  React.PropsWithChildren<unknown>
```
Comment (yuchenshi) [inline src/components/Firestore/FirestoreEmulatedApiProvider.tsx:51]: Is there a reason not to do this?

```suggestion
  React.PropsWithChildren<{}>
```

File: src/components/Home/LocalWarningCallout.tsx [hunk 2]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import { useLocalStorage, writeStorage } from '@rehooks/local-storage';
+// import { useLocalStorage, writeStorage } from '@rehooks/local-storage';
```
Comment (yuchenshi) [inline src/components/Home/LocalWarningCallout.tsx:17]: Does this not work at all? If so, please add a TODO comment with link to an issue. Also, I bet we can easily implement this ourselves...

File: src/firebase.ts [hunk 3]
```diff
@@ -14,50 +14,38 @@
  * limitations under the License.
  */
 
-import 'firebase/database';
-import 'firebase/firestore';
 // Force the browser build even in Node.js + jsdom unit tests because jsdom's
```
Comment (yuchenshi) [inline src/firebase.ts:17]: I think we no longer need this hack

File: package.json [hunk 4]
```diff
@@ -60,20 +64,22 @@
     "react-router-dom": "^5.3.0",
     "react-router-dom-v5-compat": "^6.3.0",
     "react-scripts": "5.0.1",
```
Comment (jhuleatt) [inline package.json:66]: I think this can be removed now that we're on Vite

File: src/components/Home/LocalWarningCallout.tsx (hunk 2 above)
Comment (tjlav5) [reply src/components/Home/LocalWarningCallout.tsx:17]: Reverted - fixed w/ update

File: src/components/Database/testing/DatabaseTestProviders.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+import { DatabaseEmulatedApiProvider } from '../DatabaseEmulatedApiProvider';
+
+interface RenderOptions {
+  path?: string;
+  state?: Partial<unknown>;
+  namespace?: string;
+}
+
+export const renderWithDatabase = async (
+  children: (database: Database) => Promise<React.ReactElement>,
```
[the remark this answers is not in the document]
Comment (tjlav5) [reply src/components/Database/testing/DatabaseTestProviders.tsx:37]: Ack - synced offline and I think we're ok revisiting this in a quick follow up to unblock the repo.

File: src/components/Storage/providers/StorageFirebaseAppProvider.tsx [hunk 6]
```diff
@@ -14,34 +14,57 @@
...
+} from 'firebase/storage';
+import React, { useCallback, useState } from 'react';
+import { FirebaseAppProvider, StorageProvider } from 'reactfire';
 
 import { useEmulatedFirebaseApp } from '../../../firebase';
 import { useEmulatorConfig } from '../../common/EmulatorConfigProvider';
 
-const FIREBASE_APP_OPTIONS = {};
+const FIREBASE_APP_OPTIONS = {
+  storageBucket: 'foo.appspot.com',
```
Comment (yuchenshi) [inline src/components/Storage/providers/StorageFirebaseAppProvider.tsx:30]:
```suggestion
  storageBucket: 'demo-example.appspot.com',
```

File: src/components/Storage/providers/StorageFirebaseAppProvider.tsx [hunk 7]
```diff
@@ -14,34 +14,57 @@
...
+  const [storage, setStorage] = useState<FirebaseStorage | null>(null);
   const app = useEmulatedFirebaseApp(
     'storage',
     FIREBASE_APP_OPTIONS,
     useCallback(
-      (app: any) => {
-        app.storage().useEmulator(host, port);
+      (app: FirebaseApp) => {
+        const storage = bucket
+          ? getStorage(app, `gs://other.appspot.com`)
```
Comment (yuchenshi) [inline src/components/Storage/providers/StorageFirebaseAppProvider.tsx:44]:
```suggestion
          ? getStorage(app, `gs://demo-example2.appspot.com`)
```

File: src/components/Database/DataViewer/NodeLink.test.tsx [hunk 8]
```diff
@@ -15,45 +15,47 @@
...
-  );
-  (ROOT_REF.toString as jest.Mock).mockReturnValue('http://localhost:9000/');
-});
-
-it('renders a link with the key name', () => {
-  const { getByText } = render(
-    <MemoryRouter initialEntries={['/database/test/data/a/b']}>
-      <NodeLink dbRef={REF} />
-    </MemoryRouter>
+// const ROOT_REF = fakeReference({ key: null, parent: null });
```
[the remark this answers is not in the document]
Comment (yuchenshi) [reply src/components/Database/DataViewer/NodeLink.test.tsx:25]: bump

**Review by yuchenshi — APPROVED** [review]

**Pull request merged**